crates/typst-library/src/text/raw.rs

```diff
@@ -247,7 +262,34 @@ impl Synthesize for RawElem {
 impl Show for RawElem {
     #[tracing::instrument(name = "RawElem::show", skip_all)]
     fn show(&self, _: &mut Vt, styles: StyleChain) -> SourceResult<Content> {
-        let text = self.text();
+        let mut text = self.text();
+        if text.contains('\t') {
+            let tab_size = RawElem::tab_size_in(styles);
+            let amount = text.chars().filter(|&c| c == '\t').count();
+            let mut res =
+                EcoString::with_capacity(text.len() - amount + amount * tab_size);
+            let replacement = " ".repeat(tab_size);
+            let divisor = tab_size.max(1);
+            let mut column = 0;
+            for c in text.chars() {
```

Comment:
It's an interesting question whether we want chars() or graphemes() here. Multi-char graphemes will probably typically mess with monospace anyway.

Reply:
After testing of the graphene example from the docs: `a̐éö̲` is 4 wide in the pdf, `.chars()` thinks 5 and `.graphens(true/false)` thinks 3. I changed it to `.graphenes(true)`, because it should be correct if `.chars()` if correct and sometimes better.